### ISSUE TITLE
Ranking prediction are all same value 

### ISSUE THREAD

**Issue by 1a1a11a** [dmlc/xgboost#6955]
I am using the XGBoost C API to do some ranking tasks and I have tried to use `rank:pairwise`, `rank:map`, and `rank:ndcg` as the objective, however, when I use `rank:map`, and `rank:ndcg` to train and predict, all the predicted values are constant, but it is not a problem when I use `rank:pairwise`. Does anyone know what could be wrong in my setup? Or is there a better way other than digging into the source code? Thank you!

**Comment by 1a1a11a** [comment]
I have checked that the relevance in the training data are all from 0 to 1, and the inference data are not constant. I think the training has some problems, the metric does not change at all from the first iteration to 10 iterations (while it changes if I use 
regression or rank:pairwise.

**Comment by creat89** [comment]
For NDCG, and I'm guessing MAP too, the relevance scores must be integers. See:

https://github.com/dmlc/xgboost/issues/6352#issuecomment-735595924

or:

https://discuss.xgboost.ai/t/very-large-ndcg-result/1712
